core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -85,9 +93,14 @@ public void run(final ActionContext context) throws Exception {
 
       LOG.debug("Output: {}", out);
       LOG.error("Errors: {}", err);
+      Integer exitCode = session.getExitStatus();
+      if (exitCode != null && exitCode > 0) {
+        throw new IOException(String.format("Error: command finished with exit code: %d", exitCode));
+      }
 
       if (config.output != null) {
-        context.getArguments().set(config.output, out);
+        //regex removes the carriage return at the end of the line
+        context.getArguments().set(config.output, out.replaceAll("\\s+$", ""));
```

Comment:
Removes carriage return and any other whitespace characters at the end of the output. Expected?